frontend/src/citizen-frontend/calendar/HolidayModal.tsx

```diff
@@ -22,17 +84,33 @@ export const HolidayModal = React.memo(function HolidayModal({
   close,
   reload,
   availableChildren
-}: Props) {
+}: HolidayModalProps) {
   const i18n = useTranslation()
   const [lang] = useLang()
   const { holidayPeriods } = useHolidayPeriods()
 
+  const [form, setForm] = useState<HolidayForm>(() =>
+    initializeForm(availableChildren)
+  )
+
+  const selectFreePeriod = useCallback(
+    (childId: string, dateRange: FiniteDateRange | null) => {
+      setForm((previous) => ({
+        selections: {
+          ...previous.selections,
+          [childId]: { selectedFreePeriod: dateRange }
+        }
+      }))
```

Comment:
Nitpicking, but this might be prone to errors if more fields are added to `HolidayForm` (due to missing top level spread).
Maybe `HolidayForm` could simply be `type HolidayForm = Record<string, SelectedHolidays>`.

Reply:
To refactor when adding more fields here.